crates/rome_js_formatter/src/js/expressions/assignment_expression.rs

```diff
@@ -16,11 +14,19 @@ impl ToFormatElement for JsAssignmentExpression {
             right,
         } = self.as_fields();
 
+        let right = right?;
+
+        let right = if matches!(right, JsAnyExpression::JsAssignmentExpression(_)) {
+            soft_line_indent_or_space(right.format(formatter)?)
+        } else {
+            format_elements![space_token(), right.format(formatter)?]
+        };
+
```

Comment:
Nit: You can write this as 
```suggestion
        let right = right?;

        let right = if let JsAnyExpression::JsAnyAssignmentExpression(right)) = right?  {
            soft_line_indent_or_space(right.format(formatter)?)
        } else {
            format_elements![space_token(), right.format(formatter)?]
        };

```

Reply:
No I can't because `right` gets moved and can't be inside the `else` branch